net/tcp/tcp_input.c

```diff
@@ -628,6 +628,37 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
 
   dev->d_len -= (len + iplen);
 
+#if defined(CONFIG_NET_STATISTICS) && \
+    defined(CONFIG_NET_TCP_DEBUG_DROP_RECV)
+
+#pragma message \
+  "CONFIG_NET_TCP_DEBUG_DROP_RECV is selected, this is debug " \
+  "feature to drop the tcp received packet on the floor, " \
+  "please confirm the configuration again if you do not want " \
+  "debug the TCP stack."
+
+  /* Debug feature to drop the tcp received packet on the floor */
+
+  if (dev->d_len > 0)
+    {
+      if ((g_netstats.tcp.recv %
+          CONFIG_NET_TCP_DEBUG_DROP_RECV_PROBABILITY) == 0)
```

Comment:
It is not a probability :) but dropping each `CONFIG_NET_TCP_DEBUG_DROP_RECV_PROBABILITY`s packet.
Maybe we can implement probability function as an improvement

Reply:
emm... the packet drop rate is controlled within the define of probabillity, if it is set to 50, every 50 packets will be drop one, I do not free there is anything wrong with _PROBABILITY, or maybe we need change the config name _PROBABILITY to _INTERVAL ?

Reply:
We can keep "probability", but keep in mind that it is "interval". When I read "probability" I was expecting to see probability function based on `rand` and not `a % b`. We can change condition later.

Reply:
Got!